lib/plugins/healthcheck/commonHealthcheckPlugin.ts

```diff
@@ -94,17 +95,16 @@ function addRoute(
 
       if (opts.healthChecks.length) {
         const results = await Promise.all(
-          opts.healthChecks.map((healthcheck) => {
-            return healthcheck.checker(app).then((result) => {
-              if (result.error) {
-                app.log.error(result.error, `${healthcheck.name} healthcheck has failed`)
-              }
-              return {
-                name: healthcheck.name,
-                result,
-                isMandatory: healthcheck.isMandatory,
-              }
-            })
+          opts.healthChecks.map(async (healthcheck) => {
```

Comment:
why add await? it's generally discouraged to mix callbacks and async/await

Reply:
Thanks to it, the code is less nested and easier to read compared to using `then`.
I'd agree that mixing async/await with than/catch should be avoided, but I haven't found any issues with async/await and callbacks.

Reply:
it's not a complicated code and is unlikely to change much, if ever.

it causes problems if you throw an error within that block, you'll get an unhandled rejection

Reply:
There is an entire talk from Node.js core maintainers about why it shouldn't be done, I believe I've shared it in the past